### PR TITLE
updated FacadeBase

### PR DESCRIPTION
**Pull request by brtkkndrl**
implemented Exists method in FacadeBase,
Changed GetById to return nullable and affected tests

### CODE REVIEW

**Review by KoubaFromPucks — APPROVED** [review]

File: src/Quizzer.API.BL/Facades/FacadeBase.cs [hunk 1]
```diff
@@ -19,16 +19,25 @@ public abstract class FacadeBase<TEntity, TDetailModel> : IFacade<TEntity, TDeta
...
             using IUnitOfWork uow = p_unitOfWorkFactory.Create();
 
             using IRepository<TEntity> repo = uow.GetRepository<TEntity>();
 
-            TEntity? entites = repo.GetById(id, IncludesNavigationPathDetails);
+            TEntity? entity = repo.GetById(id, IncludesNavigationPathDetails);
 
-            return p_mapper.Map<TDetailModel>(entites);
+            if (entity is null)
+                return null;
```
Comment (vecerap) [inline src/Quizzer.API.BL/Facades/FacadeBase.cs]: I would avoid placing the expression on the newline without braces, either leave it on single line or add braces

File: src/Quizzer.API.BL/Facades/QuestionFacade.cs [hunk 2]
```diff
@@ -18,8 +18,11 @@ public class QuestionFacade : FacadeBase<QuestionEntity, QuestionDetailModel> {
             $"{nameof(QuestionEntity.UserAnswers)}"
         };
 
-        public override QuestionDetailModel GetDetailById(Guid id) {
-            QuestionDetailModel model = base.GetDetailById(id);
+        public override QuestionDetailModel? GetDetailById(Guid id) {
+            QuestionDetailModel? model = base.GetDetailById(id);
+
+            if (model is null)
+                return null;
```
Comment (vecerap) [inline src/Quizzer.API.BL/Facades/QuestionFacade.cs]: The same is here

Comment (brtkkndrl) [reply src/Quizzer.API.BL/Facades/QuestionFacade.cs]: fixed

File: src/Quizzer.API.BL/Facades/FacadeBase.cs (hunk 1 above)
Comment (brtkkndrl) [reply src/Quizzer.API.BL/Facades/FacadeBase.cs]: fixed

**Review by vecerap — APPROVED** [review]
Squash

**Comment by vecerap** [comment]
Did you sqash? @BRTEK007

**Comment by brtkkndrl** [comment]
> Did you sqash? @BRTEK007

squashed @vecerap

**Pull request merged**